pkg/entry/mongo/document.go

```diff
@@ -147,6 +148,21 @@ func (d *LogDocument) Populate(ctx context.Context, ts time.Time, record map[int
 	}
 	d.Log = cleanLogContent(logContent)
 
+	orchestration, err := parse.ExtractStringValue(record, OrchestrationKey)
```

Comment:
I suggest to write new function `parse.ExtractBooleanValue()`?

[`strconv`](https://pkg.go.dev/strconv) package will help. ;-)

Reply:
Second suggestion, instead of a boolean, this can be kind of enumeration. I would rename the field `source`:
```
- user
- argo
- saagie
- minio
- ...
```

Of course, only the 2 first will be used in the first stage :-)

Reply:
I'have changed this part completely due to functional changes. Please have a look